### ISSUE TITLE
Godot editor build doesn't connect to WebSocket server (client in HTML5 works)

### ISSUE THREAD

**Issue by berk-can** [godotengine/godot#46972]
**Godot version:**
3.2.3 stable

**OS/device including version:**
Windows GLES2

**Issue description:**
When I run godot project via editor it doesnt do anything related to websocket but when I export it to HTML5 and run it then same code can connect to websocket. What might the reason be?
PS: I am trying everything in same PC

**Steps to reproduce:**
Start a websocket server, code websocket client on godot, run in editor it does nothing

**Comment by Calinou** [comment]
@berk-can Please upload a [minimal reproduction project](https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#provide-a-simple-example-project) to make this easier to troubleshoot.

Also, which program are you using to host the WebSocket server?

**Comment by berk-can** [comment]
I am using go project as websocket server (Note: I cant even get a connection log triggered on server when tried to connect from Godot editor)
Code I try is this:

```
extends Node

onready var Utils=preload("util/logger/logger.gd");

var url="ws://localhost:8080/echo"
var _client = WebSocketClient.new()
var _write_mode = WebSocketPeer.WRITE_MODE_BINARY
var _use_multiplayer = true
var last_connected_client = 0

func _init():
	_client.connect("connection_established", self, "_client_connected")
	_client.connect("connection_error", self, "_client_disconnected")
	_client.connect("connection_closed", self, "_client_disconnected")
	_client.connect("server_close_request", self, "_client_close_request")

	_client.connect("peer_packet", self, "_client_received")
	_client.connect("peer_connected", self, "_peer_connected")
	_client.connect("connection_succeeded", self, "_client_connected", ["multiplayer_protocol"])
	_client.connect("connection_failed", self, "_client_disconnected")

func _client_close_request(code, reason):
	Utils._log(_log_dest, "Close code: %d, reason: %s" % [code, reason])

func _peer_connected(id):
	Utils._log(_log_dest, "%s: Client just connected" % id)
	last_connected_client = id

func _exit_tree():
	_client.disconnect_from_host(1001, "Bye bye!")

func _process(delta):
	if _client.get_connection_status() == WebSocketClient.CONNECTION_DISCONNECTED:
		return

	_client.poll()

func _client_connected(protocol=""):
	Utils._log(_log_dest, "Client just connected with protocol: %s" % protocol)
	_client.get_peer(1).set_write_mode(_write_mode)

func _client_disconnected(clean=true):
	Utils._log(_log_dest, "Client just disconnected. Was clean: %s" % clean)


func connect_to_url():
	return _client.connect_to_url(url)

func disconnect_from_host():
	_client.disconnect_from_host(1000, "Bye bye!")

func send_data(data, dest):
	_client.get_peer(1).set_write_mode(_write_mode)
	if _use_multiplayer:
		_client.set_target_peer(dest)
		_client.put_packet(Utils.encode_data(data, _write_mode))
	else:
		_client.get_peer(1).put_packet(Utils.encode_data(data, _write_mode))

func set_write_mode(mode):
	_write_mode = mode
```

**Comment by Faless** [comment]
Likely the usual dual stack IPv4/IPv6 issue and `localhost` being resolved as `::1` while the server listens on `127.0.0.1`.
See #45249 and https://github.com/godotengine/godot/issues/36418#issuecomment-589963787 .
